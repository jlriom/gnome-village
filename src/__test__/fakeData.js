export const category = {
    category: "category",
    count: 0
}

export const habitantsByHairColor = [
    {
        hair_color: 'Black',
        count: 256
    },
    {
        hair_color: 'Gray',
        count: 258
    },
    {
        hair_color: 'Green',
        count: 267
    },
    {
        hair_color: 'Pink',
        count: 285
    },
    {
        hair_color: 'Red',
        count: 273
    }
];

export const habitantsByProfession = [
    {
        profession: 'Baker',
        count: 214
    },
    {
        profession: 'Blacksmith',
        count: 180
    },
    {
        profession: 'Brewer',
        count: 200
    },
    {
        profession: 'Butcher',
        count: 210
    },
    {
        profession: 'Carpenter',
        count: 216
    },
    {
        profession: 'Cook',
        count: 190
    },
    {
        profession: 'Farmer',
        count: 205
    },
    {
        profession: 'Gemcutter',
        count: 186
    },
    {
        profession: 'Leatherworker',
        count: 185
    },
    {
        profession: 'Marble Carver',
        count: 190
    },
    {
        profession: 'Mason',
        count: 214
    },
    {
        profession: 'Mechanic',
        count: 182
    },
    {
        profession: 'Medic',
        count: 187
    },
    {
        profession: 'Metalworker',
        count: 192
    },
    {
        profession: 'Miner',
        count: 201
    },
    {
        profession: 'Potter',
        count: 202
    },
    {
        profession: 'Prospector',
        count: 202
    },
    {
        profession: 'Sculptor',
        count: 182
    },
    {
        profession: 'Smelter',
        count: 203
    },
    {
        profession: 'Stonecarver',
        count: 187
    },
    {
        profession: 'Tailor',
        count: 199
    },
    {
        profession: 'Tax inspector',
        count: 192
    },
    {
        profession: ' Tinker',
        count: 202
    },
    {
        profession: 'Woodcarver',
        count: 199
    }
];

export const auth = { name: "name", isGuest: false };

export const habitant = {
    id: 362,
    name: 'Cogwitz Bittergrill',
    thumbnail: 'http://www.publicdomainpictures.net/pictures/120000/nahled/white-hen.jpg',
    age: 127,
    weight: 40.074535,
    height: 93.66314,
    hair_color: 'Black',
    professions: [
        'Prospector',
        ' Tinker',
        'Baker',
        'Tailor',
        'Leatherworker',
        'Sculptor'
    ],
    friends: [
        'Whitwright Fussbuster',
        'Fizwood Bitterrocket',
        'Fizwood Felpower'
    ]
};

export const friends = [
    {
        id: 363,
        name: 'Cogwitzs friend ',
        thumbnail: 'http://www.publicdomainpictures.net/pictures/120000/nahled/white-hen.jpg',
        age: 127,
        weight: 40.074535,
        height: 93.66314,
        hair_color: 'Black',
        professions: [
            'Prospector',
            ' Tinker'
        ],
        friends: [
            'Whitwright Fussbuster'
        ]
    }
];
