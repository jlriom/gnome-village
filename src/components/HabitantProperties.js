import React from "react";
import PropTypes from "prop-types";

const RenderHabitantProperty = ({ name, value }) => (
	<>
		<dt className='col-sm-6'> {name}</dt>
		<dd className='col-sm-6'> {value}</dd>
	</>
);

RenderHabitantProperty.propTypes = {
	name: PropTypes.string.isRequired,
	value: PropTypes.any.isRequired
};

const HabitantProperties = ({ age, hair_color, weight, height }) => (
	<>
		<RenderHabitantProperty name='Age:' value={age} />
		<RenderHabitantProperty name='Hair Color:' value={hair_color} />
		<RenderHabitantProperty name='Weight:' value={weight.toFixed(3)} />
		<RenderHabitantProperty name='Height:' value={height.toFixed(3)} />
	</>
);

HabitantProperties.propTypes = {
	age: PropTypes.number.isRequired,
	hair_color: PropTypes.string.isRequired,
	weight: PropTypes.number.isRequired,
	height: PropTypes.number.isRequired
};

export default HabitantProperties;
