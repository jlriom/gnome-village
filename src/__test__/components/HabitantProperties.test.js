import React from "react";
import HabitantProperties from "../../components/HabitantProperties";
import renderer from "react-test-renderer";
import { habitant } from "../fakeData";

describe("<HabitantProperties />", () => {
    it("renders ok", () => {
        const tree = renderer.create(
            <HabitantProperties {...habitant} />
        );
        expect(tree).toMatchSnapshot();
    });
});
