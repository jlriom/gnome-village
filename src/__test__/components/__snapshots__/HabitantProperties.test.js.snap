// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HabitantProperties /> renders ok 1`] = `
Array [
  <dt
    className="col-sm-6"
  >
     
    Age:
  </dt>,
  <dd
    className="col-sm-6"
  >
     
    127
  </dd>,
  <dt
    className="col-sm-6"
  >
     
    Hair Color:
  </dt>,
  <dd
    className="col-sm-6"
  >
     
    Black
  </dd>,
  <dt
    className="col-sm-6"
  >
     
    Weight:
  </dt>,
  <dd
    className="col-sm-6"
  >
     
    40.075
  </dd>,
  <dt
    className="col-sm-6"
  >
     
    Height:
  </dt>,
  <dd
    className="col-sm-6"
  >
     
    93.663
  </dd>,
]
`;
